BST.rb

```diff
@@ -0,0 +1,411 @@
+
+
+class TreeNode
+
+    attr_accessor :value, :left, :right
+
+    def initialize(value = nil, left = nil, right = nil)
+      @value = value
+      @left = left
+      @right = right
+    end
+end
+
+
+class BinarySearchTree
+    
+    def initialize()
+        @root = nil
+    end
+
+    def insert(value)
+        if @root == nil
+            @root = TreeNode.new(value)
+        else
+            curr_node = @root
+            previous_node = @root
+
+            while curr_node != nil
+                previous_node = curr_node
+                if value <= curr_node.value
+                    curr_node = curr_node.left
+                else
+                    curr_node = curr_node.right
+                end
+            end
+
+            if value < previous_node.value
+                previous_node.left = TreeNode.new(value)
+            else
+                previous_node.right = TreeNode.new(value)
+            end
+        end
+    end
+
+
+    def traverse        
+        print "Inorder traversal => "
+        in_order(@root)
+        puts
+
+        print "Post-Order traversal => "
+        post_order(@root)
+        puts
+
+        print "Pre-Order traversal => "
+        pre_order(@root)
+        puts
+
+        print "Level-Order traversal => "
+        level_order(@root)
+        puts
+    end
+
+
+    def in_order(root)
+        return if root.nil?
+            
+        in_order(root.left)
+
```

Comment:
Spacing not needed here, rethink spacing in the whole file.
Space should be used to separate logically different parts.